ui/desktop/tests/e2e/authentication.spec.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+/* eslint-disable no-undef */
+const { test } = require('./fixtures/baseTest');
+const LoginPage = require('./pages/loginPage');
+const { expect } = require('@playwright/test');
+
+/* eslint-disable no-undef */
+test.describe('user/password authentication tests', async () => {
+  test('Authenticates and signs out', async ({
+    electronPage,
+    clusterUrl,
+    username,
+    password,
+  }) => {
+    const loginPage = new LoginPage(electronPage);
```

Comment:
Question for learning:
We setup this `LoginPage` so we can use the predefined methods? However, it still points to the `electronPage` that we make our assertions against?

Reply:
Correct, I could have also called it `electronWindow` but it returns an actual `page` object in the end so that's what we're actually working in and asserting on. The `LoginPage` is meant to encapsulate the login screen itself, and we might have something like `TargetsPage` or `SettingsPage` that all still use the `electronPage` (more realistically `authedPage`) to handle common actions on that specific page.